### ISSUE TITLE
Don't know how to load github secret values into dockerfile's env and use them in node.js app

### ISSUE THREAD

**Issue by ywbyeol** [docker/build-push-action#788]
### Troubleshooting

Before submitting a bug report please read the [Troubleshooting doc](https://github.com/docker/build-push-action/blob/master/TROUBLESHOOTING.md).

### Behaviour

I've written a workflow that builds a docker image and pushes it to docker hub when pushing a javascript file to a repository, but I'm running into a problem.

The action itself that builds the docker image and pushes it to the docker hub seems to be working fine, but I think I've mis-written the logic to import the secret values from the yaml file into the docker file and allocate them so that they can be used as env, but I don't know how to solve it.

#### Expected behaviour

I would like to receive the TOKEN and CLIENTID stored in GitHub secrets and store them as variables in the index.js file to be executed in the docker image. 

#### Actual behaviour

 But when I build and run the docker image, no values are assigned to TOKEN and CLIENTID.
![docker](https://user-images.githubusercontent.com/90835115/216142914-eb003e55-8bae-4909-8be6-2b57e4b4d158.png)

#### Steps to reproduce this issue

[Here's my repo address](https://github.com/ywbyeol/discord-bot/)
Since I'm new to docker and github actions, the commits and code feel cluttered. I apologize for this and hope you think of it as cute.

### Configuration

```yml
name: Build and Push Docker Image

on:
  push:
    branches:
      - main
    paths:
      - '**.js'

jobs:
  build-and-push-docker-image:
    runs-on: ubuntu-latest
    steps:

      - name: Set up QEMU
        uses: docker/setup-qemu-action@v2

      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2

      - name: Login to Docker Hub

        uses: docker/login-action@v2
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}

      - name: Build and push
        uses: docker/build-push-action@v4
        with:
          push: true
          tags: ${{ secrets.DOCKERHUB_USERNAME }}/discord-bot:latest
          build-args: |
            TOKEN=${{ secrets.TOKEN }}
            CLIENTID=${{ secrets.CLIENTID }}
```

```Dockerfile
# syntax=docker/dockerfile:1
FROM node:latest

WORKDIR /app

COPY package*.json ./

# RUN --mount=type=secret,id=TOKEN \
#     --mount=type=secret,id=CLIENTID \
#     export TOKEN=$(cat /run/secrets/TOKEN) && \
#     export CLIENTID=$(cat /run/secrets/CLIENTID) && \
#     npm ci

RUN npm ci

COPY src/ ./src/

ENV TOKEN=$TOKEN
ENV CLIENTID=$CLIENTID

RUN node src/deploy-commands.js

CMD [ "node", "src/index.js" ]
```

```js
const token = process.env.TOKEN;
```

### Logs

[Here's my repo actions address](https://github.com/ywbyeol/discord-bot/actions)
</br></br></br>

**Comment by NikolaSivkov** [comment]
Don't store the secrets in the container, pass them as env variables when you deploy the image. like this : 
```
docker run -d -e TOKEN="myToken" -e CLIENTID="myclientId" mycontainer/image:tag
```
or 

from a secure (only your user has read access) file like this : 

```
$ docker run --env-file ./env.list mycontainer/image:tag
```
Where the file is formatted like this

```
TOKEN="myToken"
CLIENTID="myclientId"
```

**Comment by ywbyeol** [comment]
> 

Thank you so much for your advice. 

Before registering the question, I tried to pass the environment variable with -e a couple of times, but it probably failed due to

my lack of understanding. Thanks to you, I got a sense of what to study in the documentation, and after several attempts, I built 

a successful CI/CD that can be deployed automatically to gce. Now I can just focus on developing new features.

![success](https://user-images.githubusercontent.com/90835115/216279030-17c2e8fe-33c9-492f-95ef-8d1ade458799.png)

Thanks again!
</br></br>
